core/src/main/java/com/datastax/oss/driver/internal/core/pool/ChannelPool.java

```diff
@@ -167,18 +167,23 @@ public DriverChannel next() {
    *     to the caller to fail fast and move to the next node.
    *     <p>There is no need to return the channel.
    */
-  public DriverChannel next(@Nullable Token routingKey) {
+  public DriverChannel next(@Nullable Token routingKey, @Nullable Integer shardSuggestion) {
     if (!singleThreaded.initialized) {
       return null;
     }
     if (singleThreaded.shardingInfo == null) {
       return channels[0].next();
     }
 
-    int shardId =
-        routingKey != null
-            ? singleThreaded.shardingInfo.shardId(routingKey)
-            : ThreadLocalRandom.current().nextInt(channels.length);
+    int shardId;
+    if (shardSuggestion != null && shardSuggestion < channels.length) {
+      shardId = shardSuggestion;
```

Comment:
`ShardSuggestion` being null is a much different condition than it exceeding length of `channels`.
In the second case I'd expect at least a warning.

Reply:
I've wondered if i can skip this check entirely, because it should not really happen unless someone changes node configuration so that specifically this does happen, but I've added the warning.